plugins/reporters/scanoss/src/main/resources/templates/asciidoc/scanoss_snippet.ftl

```diff
@@ -0,0 +1,141 @@
+[#--
```

Comment:
I'm not the expert here, but do we really need a separate snippet report for SCANOSS? Can't we generalize the existing `FossIdSnippetReporter` so that it works for any snippet scanner?

I'd like to avoid that each new snippet scanner we introduce support for also requires its own report.

Reply:
The separate scanner reports approach was discussed with Nikola, who preferred keeping them separate due to concerns about diverging functionality.

However, you make a valid point. Perhaps having a generic reporter that can adapt to each scanner's particularities would be better? @nnobelis, what are your thoughts on this?

Reply:
@sschuberth :

While it is true that the templates are nearly the same, they are still some differences between FossID and ScanOSS.
Therefore I would like to avoid a bunch of if/else statements in them and keep two different Freemarker templates. 

This also make the plugins self-contained, no ?
If they were to share the templates, where the latter would be located ?

Additionally, The reporters themselves seems to be identical, so maybe they could be merged but the same problem would arise: where the class should be located in ORT ?

Reply:
> The reporters themselves seems to be identical, so maybe they could be merged but the same problem would arise: where the class should be located in ORT ?

Well, quite simply, we could have a generic "snippet scanner reporter" in ORT that neither carries "FossID nor "SCANOSS" in its name, but only mentions the used / configured scanner as part of the generated report.

Reply:
Understood. So we can do that for the Reporter class but as said before, I would like to keep two different templates.

But this is again increasing the complexity of this PR so would you agree to keep this "shared-reporter" approach in a follow-up PR ?

Reply:
> I would like to keep two different templates.

I'd agreed to that if it turns out that we really would need too many conditionals in the template. But I'm bot yet convinced of that.

> But this is again increasing the complexity of this PR so would you agree to keep this "shared-reporter" approach in a follow-up PR ?

Yes, see [my wish here](https://github.com/oss-review-toolkit/ort/pull/10004#discussion_r2058324292) to split this PR.